packages/workflow/src/records/fhir.ts

```diff
@@ -807,7 +885,130 @@ export async function sendBundleToHearth(
     )
   }
 
-  return res.json()
+  const responseBundle: TransactionResponse = await res.json()
+
+  const ok = responseBundle.entry.every((e) =>
+    e.response.status.startsWith('2')
+  )
+  if (!ok) {
+    throw new Error(
+      'Hearth was unable to create/update all the entires in the bundle'
+    )
+  }
+
+  return responseBundle
+}
+
+function findSavedReference(
+  temporaryReference: URNReference,
+  resourceBundle: Bundle,
+  responseBundle: TransactionResponse
+): URLReference | null {
+  const indexInResponseBundle = resourceBundle.entry.findIndex(
+    (entry) => entry.fullUrl === temporaryReference
+  )
+  if (indexInResponseBundle === -1) {
+    return null
+  }
+  return responseBundle.entry[indexInResponseBundle].response.location
+}
+
+function toSavedComposition(
+  composition: Composition,
+  id: UUID,
+  resourceBundle: Bundle,
+  responseBundle: TransactionResponse
+): SavedComposition {
+  return {
+    ...composition,
+    id,
+    section: composition.section.map((section) => ({
+      ...section,
+      entry: section.entry.map((sectionEntry) => {
+        if (isURLReference(sectionEntry.reference)) {
+          return {
+            ...sectionEntry,
+            reference: sectionEntry.reference
+          }
+        }
+        const savedReference = findSavedReference(
+          sectionEntry.reference,
+          resourceBundle,
+          responseBundle
+        )
+        if (!savedReference) {
+          throw internal(
```

Comment:
```suggestion
          throw new Error(
```

Maybe better to just throw generic errors from this abstraction level so http related concepts are kept on the handler-level.